roles/kubernetes-apps/helm/tasks/main.yml

```diff
@@ -1,131 +1,34 @@
-    {% if tiller_secure_release_info %} --override 'spec.template.spec.containers[0].command'='{/tiller,--storage=secret}' {% endif %}
-    --override spec.selector.matchLabels.'name'='tiller',spec.selector.matchLabels.'app'='helm'
-    {% if tiller_wait %} --wait{% endif %}
-    {% if tiller_replicas is defined %} --replicas {{ tiller_replicas | int }}{% endif %}
-    --output yaml
-    | sed 's@apiVersion: extensions/v1beta1@apiVersion: apps/v1@'
-    | {{ bin_dir }}/kubectl apply -f -
-  args:
-    executable: /bin/bash
-  register: install_helm
-  when:
-    - inventory_hostname == groups['kube-master'][0]
-    - helm_version is version('v3.0.0', '<')
-  changed_when: false
-  environment: "{{ proxy_env }}"
-
-# FIXME: https://github.com/helm/helm/issues/4063
-- name: Helm | Force apply tiller overrides if necessary
-  shell: >
-    set -o pipefail &&
-    {{ bin_dir }}/helm init --upgrade --tiller-image={{ tiller_image_repo }}:{{ tiller_image_tag }} --tiller-namespace={{ tiller_namespace }}
-    {% if helm_skip_refresh %} --skip-refresh{% endif %}
-    {% if helm_stable_repo_url is defined %} --stable-repo-url {{ helm_stable_repo_url }}{% endif %}
-    {% if rbac_enabled %} --service-account={{ tiller_service_account }}{% endif %}
-    {% if tiller_node_selectors is defined %} --node-selectors {{ tiller_node_selectors }}{% endif %}
-    --override spec.template.spec.priorityClassName={% if tiller_namespace == 'kube-system' %}system-cluster-critical{% else %}k8s-cluster-critical{% endif %}
-    {% if tiller_override is defined and tiller_override %} --override {{ tiller_override }}{% endif %}
-    {% if tiller_max_history is defined %} --history-max={{ tiller_max_history }}{% endif %}
-    {% if tiller_enable_tls %} --tiller-tls --tiller-tls-verify --tiller-tls-cert={{ tiller_tls_cert }} --tiller-tls-key={{ tiller_tls_key }} --tls-ca-cert={{ tiller_tls_ca_cert }} {% endif %}
-    {% if tiller_secure_release_info %} --override 'spec.template.spec.containers[0].command'='{/tiller,--storage=secret}' {% endif %}
-    --override spec.selector.matchLabels.'name'='tiller',spec.selector.matchLabels.'app'='helm'
-    {% if tiller_wait %} --wait{% endif %}
-    {% if tiller_replicas is defined %} --replicas {{ tiller_replicas | int }}{% endif %}
-    --output yaml
-    | sed 's@apiVersion: extensions/v1beta1@apiVersion: apps/v1@'
-    | {{ bin_dir }}/kubectl apply -f -
-  args:
-    executable: /bin/bash
-  changed_when: false
-  when:
-    - inventory_hostname == groups['kube-master'][0]
-    - helm_version is version('v3.0.0', '<')
-  environment: "{{ proxy_env }}"
-
-- name: Helm | Add/update stable repo on all masters
-  command: "{{ bin_dir }}/helm repo add stable {{ helm_stable_repo_url }}"
-  environment: "{{ proxy_env }}"
-  when:
-    - helm_version is version('v3.0.0', '>=')
-    - helm_stable_repo_url is defined
```

Comment:
This task is for Helm v3.0.0+
Why do we need to remove this?

Reply:
stable repo is dead

Reply:
Indeed, see https://github.com/helm/charts/blob/master/README.md#deprecation-timeline

Reply:
IIUC the stable repo of Helm v3 is still available on newer stable rep as https://github.com/helm/charts/blob/master/README.md#how-do-i-enable-the-stable-repository-for-helm-3

That is the reason why the PR https://github.com/kubernetes-sigs/kubespray/pull/6867 updates the stable repo UR.

Reply:
Dead == deprecated, read only and full of out of date charts, see @EppO link.
I'm against promoting bad practices, but if that's the only blocker I can put that back.

Reply:
Ah, I misunderstood.
Newer stable repo also is deprecated as https://github.com/helm/charts/blob/master/README.md#status-of-the-project
Sorry about that.
I am ok to remove this part.

Reply:
No problem, helm team went all in for the deprecations